### PR TITLE
roc_auc_weighted metric addition

### PR DESCRIPTION
**Pull request by shreyas36**
## Related issue number
#774 

## Checks

- [ ] I've used [pre-commit](https://microsoft.github.io/FLAML/docs/Contribute#pre-commit) to lint the changes in this PR, or I've made sure [lint with flake8](https://github.com/microsoft/FLAML/blob/816a82a1155b4de4705b21a615ccdff67c6da379/.github/workflows/python-package.yml#L54-L59) output is two 0s.
- [ ] I've included any doc changes needed for https://microsoft.github.io/FLAML/. See https://microsoft.github.io/FLAML/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

**Comment by sonichi** [comment]
Please sign the contributor license agreement. It only needs to be signed once for all PRs.

File: notebook/integrate_azureml.ipynb [hunk 1]
```diff
@@ -100,7 +100,7 @@
    },
    "source": [
     "### Run FLAML\n",
-    "In the FLAML automl run configuration, users can specify the task type, time budget, error metric, learner list, whether to subsample, resampling strategy type, and so on. All these arguments have default values which will be used if users do not provide them. For example, the default ML learners of FLAML are `['lgbm', 'xgboost', 'catboost', 'rf', 'extra_tree', 'lrl1']`. "
+    "In the FLAML automl run configuration, users can specify the task type, time budget, error metric, learner list, whether to subsample, resampling strategy type, and so on. All these arguments have default values which will be used if users do not provide them. For example, the default ML learners of FLAML are `['lgbm', 'xgboost', 'catboost', 'rf', 'extra_tree', 'lrl1']`. Refer to the documentation to see the full list of argument values [here](https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML/) "
```
Comment (sonichi) [inline notebook/integrate_azureml.ipynb:103]: Move the added sentence to line 133.

File: website/docs/Examples/Integrate - Scikit-learn Pipeline.md [hunk 2]
```diff
@@ -32,14 +32,15 @@ automl_pipeline = Pipeline([
...
+
 ![png](images/pipeline.png)
 
 ### Run AutoML in the pipeline
 
 ```python
 automl_settings = {
     "time_budget": 60,  # total running time in seconds
-    "metric": "accuracy",  # primary metrics can be chosen from: ['accuracy','roc_auc', 'roc_auc_ovr', 'roc_auc_ovo', 'f1','log_loss','mae','mse','r2']
+    "metric": "accuracy",  # primary metrics can be chosen from: ['accuracy','roc_auc','roc_auc_weighted', 'roc_auc_ovr', 'roc_auc_ovo', 'f1','log_loss','mae','mse','r2']
```
Comment (sonichi) [inline website/docs/Examples/Integrate - Scikit-learn Pipeline.md:43]:
```suggestion
    "metric": "accuracy",  # primary metrics can be chosen from: ['accuracy', 'roc_auc', 'roc_auc_weighted', 'roc_auc_ovr', 'roc_auc_ovo', 'f1', 'log_loss', 'mae', 'mse', 'r2']
```

**Review by sonichi — COMMENTED** [review]
Please also update [automl_classification.ipynb](https://github.com/microsoft/FLAML/blob/main/notebook/automl_classification.ipynb)

**Comment by shreyas36** [comment]
@microsoft-github-policy-service agree

File: flaml/ml.py [hunk 3]
```diff
@@ -239,8 +242,10 @@ def sklearn_metric_loss_score(
     Args:
         metric_name: A string of the metric name, one of
             'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',
-            'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg',
-            'micro_f1', 'macro_f1'.
+            'roc_auc_ovo', 'roc_auc_weighted', 'roc_auc_ovo_weighted', 'roc_auc_ovr_weighted',
+
+            'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'.
+
```
Comment (sonichi) [inline flaml/ml.py:248]:
```suggestion
            'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'.
```

File: notebook/automl_classification.ipynb [hunk 4]
```diff
@@ -113,7 +113,8 @@
     "settings = {\n",
     "    \"time_budget\": 600,  # total running time in seconds\n",
     "    \"metric\": 'accuracy',  # can be: 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
-    "                           # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # check the documentation for full list of arguments(https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML/)\n",
```
Comment (sonichi) [inline notebook/automl_classification.ipynb:117]: Remove "can be...". Change the URL to be https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#optimization-metric

File: notebook/integrate_azureml.ipynb [hunk 5]
```diff
@@ -131,12 +131,13 @@
     "settings = {\n",
     "    \"time_budget\": 60,  # total running time in seconds\n",
     "    \"metric\": 'accuracy',  # can be: 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
-    "                           # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # check the documentation for full list of arguments(https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML/)\n",
```
Comment (sonichi) [inline notebook/integrate_azureml.ipynb:135]: Same as above.

File: website/docs/Examples/Integrate - Scikit-learn Pipeline.md [hunk 6]
```diff
@@ -32,14 +32,15 @@ automl_pipeline = Pipeline([
...
+
 ![png](images/pipeline.png)
 
 ### Run AutoML in the pipeline
 
 ```python
 automl_settings = {
     "time_budget": 60,  # total running time in seconds
-    "metric": "accuracy",  # primary metrics can be chosen from: ['accuracy','roc_auc', 'roc_auc_ovr', 'roc_auc_ovo', 'f1','log_loss','mae','mse','r2']
+    "metric": "accuracy",  # primary metrics can be chosen from: ['accuracy', 'roc_auc', 'roc_auc_weighted', 'roc_auc_ovr', 'roc_auc_ovo', 'f1', 'log_loss', 'mae', 'mse', 'r2']
```
Comment (sonichi) [inline website/docs/Examples/Integrate - Scikit-learn Pipeline.md:43]: same as above.

**Review by sonichi — COMMENTED** [review]
Looks good to me except a few minor things.

File: notebook/automl_classification.ipynb [hunk 7]
```diff
@@ -112,12 +112,13 @@
    "source": [
     "settings = {\n",
     "    \"time_budget\": 600,  # total running time in seconds\n",
-    "    \"metric\": 'accuracy',  # can be: 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
-    "                           # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "    \"metric\": 'accuracy',  # 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
+    "                        # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # check the documentation for full list of arguments(https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#optimization-metric)\n",
```
Comment (sonichi) [inline notebook/automl_classification.ipynb:117]:
```suggestion
    "    \"metric\": 'accuracy', ",
    "                        # check the documentation for options of metrics (https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#optimization-metric)\n",
```

File: notebook/integrate_azureml.ipynb [hunk 8]
```diff
@@ -130,13 +130,14 @@
    "source": [
     "settings = {\n",
     "    \"time_budget\": 60,  # total running time in seconds\n",
-    "    \"metric\": 'accuracy',  # can be: 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
-    "                           # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "    \"metric\": 'accuracy',  # 'r2', 'rmse', 'mae', 'mse', 'accuracy', 'roc_auc', 'roc_auc_ovr',\n",
+    "                        # 'roc_auc_ovo', 'log_loss', 'mape', 'f1', 'ap', 'ndcg', 'micro_f1', 'macro_f1'\n",
+    "                        # check the documentation for full list of arguments(https://microsoft.github.io/FLAML/docs/Use-Cases/Task-Oriented-AutoML#optimization-metric)\n",
```
Comment (sonichi) [inline notebook/integrate_azureml.ipynb:135]: Same as above.

**Review by sonichi — COMMENTED** [review]
Sorry that some previous comments were not clear enough. Made direct suggestions this time.

File: notebook/integrate_azureml.ipynb (hunk 8 above)
Comment (shreyas36) [reply notebook/integrate_azureml.ipynb]: Sorry I misinterpreted it. Committed proper changes now.

**Review by sonichi — APPROVED** [review]

**Review by ZviBaratz — APPROVED** [review]
LGTM. Nice addition, thank you.
